File: ABC271/C.swift
var num = Int(String(readLine()!))!
var sunuke = readLine()!.split(separator: " ").map { Int(String($0))! }

sunuke.sort { $0 < $1 }
 
var count:Int = 0

var books:Int = 0

var i:Int = 0

while i<sunuke.count {
    if(sunuke[i] == count+1){
        count+=1
    } else if (sunuke[i] == count) {
        books += 1
    } else {
        if (books >= 2) {
            books-=2
            sunuke.insert(count+1, at:i)
            count+=1
        } else if (books >= 1 && sunuke.count-i >= 1){
            books-=1
            sunuke.removeLast()
            sunuke.insert(count+1, at:i)
            count+=1
        } else if (sunuke.count-i >= 2){
            sunuke.removeLast()
            sunuke.removeLast()
            sunuke.insert(count+1, at:i)
            count+=1
        }
    }
    i+=1
}

count+=books/2

print(count)

File: ABC271/C.py
N = int(input())
a = list(map(int, input().split()))

a.sort()

check = [False] * (N)

books = 0

queue = []

for i in range(N):
  if a[i] > N:
    books+=1
  elif check[a[i]-1] == False:
    check[a[i]-1] = True
    queue.append(a[i])
  else:
    books+=1

count = 0

while count < N:
  if check[count] == True:
    count+=1
  else:
    if books>=2:
      count+=1
      books-=2
    elif books==1 and len(queue) >= 1:
      if count < queue[-1]:
        check[queue[-1]-1] = False
        queue.pop(-1)
        count+=1
        books-=1
      else:
        break
    elif books==0 and len(queue) >= 2:
      if count < queue[-1] and count < queue[-2]:
        check[queue[-1]-1] = False
        check[queue[-2]-1] = False
        queue.pop(-1)
        queue.pop(-1)
        count+=1
      else:
        break
    else:
      break

print(count)